### PR TITLE
feat(axe-core-4.4): Update axe-core@4.4.1

### PR DESCRIPTION
**Pull request by katydecorah**
#### Details

This PR will update axe-core to 4.4.1. This new version includes bug fixes, 17 false-positive fixes, 1 rule change, and 1 general improvement.

- Omit [color-contrast-enhanced, wcag146](https://github.com/dequelabs/axe-core/blob/f583c70b191513f5bd116a3ed091e937ea2cd65d/lib/rules/color-contrast-enhanced.json) in tests. Being AAA, this rule is out of scope.
- Update axe config to replace `branding` property value to be a string.
- Remove webpack configuration for fallback of the "crypto" import, the dependency was removed in axe-core.
- Remove manual typings that are now in axe-core.
- Exclude all rules tagged with `wcag2aaa`. The product does not include AAA rules and by filtering them out it will prevent `identical-links-same-purpose` from being included. In 4.4, some [rules were retagged](https://github.com/dequelabs/axe-core/commit/828e526fd06ee596df73f4825e750aad459ca75e), which removed the `best-practices` tag from this rule (the `best-practices` tag previously excluded `identical-links-same-purpose`).

##### To do

- [x] Understand how we should handle identical-links-same-purpose
- [x] Check if we can improve our code from the changes: 
    - [x] include/exclude property for AxeOptions has been enhanced 
    - [x] allow passing a NodeList to ElementContext 

##### Motivation

Feature work

##### Context

axe-core [4.4.0](https://github.com/dequelabs/axe-core/releases/tag/v4.4.0) and [4.4.1](https://github.com/dequelabs/axe-core/releases/tag/v4.4.1) release notes

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/scanner/__snapshots__/get-rule-inclusions.test.ts.snap [hunk 1]
```diff
@@ -166,8 +170,7 @@ Object {
     "status": "included",
   },
   "identical-links-same-purpose": Object {
-    "reason": "rule is tagged best-practice",
-    "status": "excluded",
+    "status": "included",
```
Comment (katydecorah) [inline src/tests/unit/tests/scanner/__snapshots__/get-rule-inclusions.test.ts.snap:173]: Do we expect [identical-links-same-purpose](https://github.com/dequelabs/axe-core/blob/f583c70b191513f5bd116a3ed091e937ea2cd65d/lib/rules/identical-links-same-purpose.json) to be included? The `best-practices` tag was removed in 4.4.0, it has the `wcag2aaa` tag, and in the tests we [declare it's omitted due to AAA](https://github.com/microsoft/accessibility-insights-web/blob/5deba6151546207ecb802540184a0c7e46947827/src/tests/unit/tests/scanner/map-axe-tags-to-guidance-links.test.ts#L17).

Comment (katydecorah) [reply src/tests/unit/tests/scanner/__snapshots__/get-rule-inclusions.test.ts.snap:173]: The team decided that the product will continue to exclude this rule.

**Review by dbjorge — APPROVED** [review]

**Comment by dbjorge** [comment]
Looks good to me!

**Review by lisli1 — APPROVED** [review]
LGTM

**Comment by dbjorge** [comment]
Merging this on @katydecorah's behalf to enable #5397 to resolve a pending merge conflict with the webpack fallback config update

**Pull request merged**